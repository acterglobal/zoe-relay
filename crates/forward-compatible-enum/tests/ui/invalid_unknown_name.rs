use forward_compatible_enum::ForwardCompatibleEnum;

#[derive(ForwardCompatibleEnum)]
#[forward_compatible(unknown_variant = "123Invalid")]  // Invalid identifier!
pub enum InvalidUnknownName {
    #[discriminant(1)]
    Variant1,
    Unknown { discriminant: u32, data: Vec<u8> },
}