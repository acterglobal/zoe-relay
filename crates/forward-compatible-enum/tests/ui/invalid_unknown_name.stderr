error: '123Invalid' is not a valid identifier
 --> tests/ui/invalid_unknown_name.rs:4:1
  |
4 | / #[forward_compatible(unknown_variant = "123Invalid")]  // Invalid identifier!
5 | | pub enum InvalidUnknownName {
6 | |     #[discriminant(1)]
7 | |     Variant1,
8 | |     Unknown { discriminant: u32, data: Vec<u8> },
9 | | }
  | |_^

error[E0601]: `main` function not found in crate `$CRATE`
 --> tests/ui/invalid_unknown_name.rs:9:2
  |
9 | }
  |  ^ consider adding a `main` function to `$DIR/tests/ui/invalid_unknown_name.rs`
